### PR TITLE
Hotfix/display 1019 feed errors

### PR DESCRIPTION
**Pull request by tuj**
#### Link to ticket

https://jira.itkdev.dk/browse/DISPLAY-1019

#### Description

- Wrapped feeds in try-catch to avoid throw errors.
- Added unpublished flow to EventDatabase feed when occurrence returns 404.
- Fixed EventDatabase feed poster subscription parameters not being applied when calling getData().

#### Checklist

- [ ] My code is covered by test cases.
- [x] My code passes our test (all our tests).
- [ ] My code passes our static analysis suite.
- [x] My code passes our continuous integration process.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,9 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+- [#166](https://github.com/os2display/display-api-service/pull/166)
+  - Wrapped feeds in try-catch to avoid throw errors.
```
Comment (rimi-itk) [inline CHANGELOG.md:8]:
```suggestion
  - Wrapped feeds in try-catch to avoid throwing errors.
```

File: src/Feed/EventDatabaseApiFeedType.php [hunk 2]
```diff
@@ -19,107 +21,116 @@ class EventDatabaseApiFeedType implements FeedTypeInterface
...
+                                ];
+                            }
+
+                            return [$eventOccurrence];
+                        }
+                }
+            }
+        } catch (\Throwable $throwable) {
+            // If the content does not exist anymore, unpublished the slide.
+            if ($throwable instanceof ClientException && 404 == $throwable->getCode()) {
```
Comment (rimi-itk) [inline src/Feed/EventDatabaseApiFeedType.php:124]:
```suggestion
            if ($throwable instanceof ClientException && Response::HTTP_NOT_FOUND == $throwable->getCode()) {
```

(cf. https://github.com/symfony/symfony/blob/6.3/src/Symfony/Component/HttpFoundation/Response.php#L51)

File: src/Feed/EventDatabaseApiFeedType.php [hunk 3]
```diff
@@ -19,107 +21,116 @@ class EventDatabaseApiFeedType implements FeedTypeInterface
...
+                try {
+                    $this->logger->info('Feed with id: '.$feed->getId().' depends on an item that does not exist in Event Database. Unpublished slide with id: '.$feed->getSlide()->getId());
```
Comment (rimi-itk) [inline src/Feed/EventDatabaseApiFeedType.php:126]: You should use placeholders in the log message: https://symfony.com/doc/5.4/logging.html#logging-a-message

And probably move the logging to _after_ unpublishing the slide (cf. “Unpublished …” (past tense)).

File: src/Feed/EventDatabaseApiFeedType.php [hunk 4]
```diff
@@ -19,107 +21,116 @@ class EventDatabaseApiFeedType implements FeedTypeInterface
...
+                    $feed->getSlide()->setPublishedTo(new \DateTime('now', new \DateTimeZone('UTC')));
```
Comment (rimi-itk) [inline src/Feed/EventDatabaseApiFeedType.php:127]: Does this _unpublish_ a slide? To the untrained eye, it seems to _publish_ a slide.

File: src/Feed/EventDatabaseApiFeedType.php [hunk 5]
```diff
@@ -19,107 +21,116 @@ class EventDatabaseApiFeedType implements FeedTypeInterface
...
+                    $this->entityManager->flush();
+                } catch (\Exception $exception) {
+                    $this->logger->error($exception->getCode().': '.$exception->getMessage());
```
Comment (rimi-itk) [inline src/Feed/EventDatabaseApiFeedType.php:130]: Use message placeholders.

File: src/Feed/EventDatabaseApiFeedType.php [hunk 6]
```diff
@@ -19,107 +21,116 @@ class EventDatabaseApiFeedType implements FeedTypeInterface
...
+                }
+            } else {
+                $this->logger->error($throwable->getCode().': '.$throwable->getMessage());
```
Comment (rimi-itk) [inline src/Feed/EventDatabaseApiFeedType.php:133]: Use message placeholders.

File: src/Feed/EventDatabaseApiFeedType.php [hunk 7]
```diff
@@ -154,97 +165,100 @@ public function getAdminFormOptions(FeedSource $feedSource): array
...
-                        'value' => $member->{'@id'},
-                    ] : $member;
                 }
-            }
 
-            return $result;
+                return $result;
+            }
+        } catch (\Throwable $throwable) {
+            $this->logger->error($throwable->getCode().': '.$throwable->getMessage());
```
Comment (rimi-itk) [inline src/Feed/EventDatabaseApiFeedType.php:261]: Use message placeholders.

**Review by rimi-itk — CHANGES_REQUESTED** [review]
Looks good. However, we should use placeholders in log messages, cf.  https://symfony.com/doc/5.4/logging.html#logging-a-message.

File: src/Feed/EventDatabaseApiFeedType.php (hunk 4 above)
Comment (tuj) [reply src/Feed/EventDatabaseApiFeedType.php:128]: Added a comment.
A slide has publishedFrom, publishedTo. By setting publishedTo to now, the slide becomes unpublished.

Comment (rimi-itk) [reply src/Feed/EventDatabaseApiFeedType.php:128]: Ah, the “To” had escaped my untrained eye …

File: src/Feed/EventDatabaseApiFeedType.php [hunk 8]
```diff
@@ -54,9 +55,9 @@ public function getData(Feed $feed): array
 
                         $queryParams = array_filter([
                             'items_per_page' => $numberOfItems,
-                            'occurrences.place.id' => $places,
-                            'organizer.id' => $organizers,
-                            'tags' => $tags,
+                            'occurrences.place.id' => array_map(function ($place) { return str_replace('/api/places/', '', $place['value']); }, $places),
+                            'organizer.id' => array_map(function ($organizer) { return str_replace('/api/organizers/', '', $organizer['value']); }, $organizers),
+                            'tags' => array_map(function ($tag) { return str_replace('/api/tags/', '', $tag['value']); }, $tags),
```
Comment (rimi-itk) [inline src/Feed/EventDatabaseApiFeedType.php:60]: https://www.php.net/manual/en/functions.arrow.php are designed for this. Consider using them, i.e.

```suggestion
                            'occurrences.place.id' => array_map(static fn($place) => str_replace('/api/places/', '', $place['value']), $places),
                            'organizer.id' => array_map(static fn($organizer) => str_replace('/api/organizers/', '', $organizer['value']), $organizers),
                            'tags' => array_map(static fn($tag) => str_replace('/api/tags/', '', $tag['value']), $tags),
```

**Review by rimi-itk — CHANGES_REQUESTED** [review]
One final fly fornication!

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**